drivers/volume/portworx/portworx.go

```diff
@@ -2227,24 +2227,14 @@ func (d *portworx) SetReplicationFactor(vol *torpedovolume.Volume, replFactor in
 		if err != nil {
 			return nil, false, err
 		}
-		quitFlag := false
-		wdt := time.After(replicationUpdateTimeout)
-		for !quitFlag && !(areRepSetsFinal(volumeInspectResponse.Volume, replFactor) && isClean(volumeInspectResponse.Volume)) {
-			select {
-			case <-wdt:
-				quitFlag = true
-			default:
-				volumeInspectResponse, err = volDriver.Inspect(d.getContext(), &api.SdkVolumeInspectRequest{VolumeId: volumeName})
-				if err != nil && errIsNotFound(err) {
-					return nil, false, err
-				} else if err != nil {
-					return nil, true, err
-				}
-				time.Sleep(defaultRetryInterval)
-			}
+		if !waitForUpdateToFinish {
+			return nil, false, nil
```

Comment:
This method should returns only error but here it is returning tuple (3 values), Is this expected?

Reply:
yes, this code block is part of DoRetryWithTimeout where it expects (out, retry, err).
Im just telling do not wait for ha-update to finish and return without retry and error if triggered successfully